frontend/src/app/app-routing.module.ts

```diff
@@ -1,7 +1,17 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { ProfessorScheduleComponent } from './pages/professor-schedule/professor-schedule.component';
+import { StudentScheduleComponent } from './pages/student-schedule/student-schedule.component';
+import { StudentListComponent } from './pages/student-list/student-list.component';
+import { StudentKardexComponent } from './pages/student-kardex/student-kardex.component';
 
-const routes: Routes = [];
+const routes: Routes = [
+  { path: 'professor/schedule', component: ProfessorScheduleComponent },
+  { path: 'student/schedule', component: StudentScheduleComponent },
+  { path: 'student/list', component: StudentListComponent },
+  { path: 'student/kardex', component: StudentKardexComponent },
```

Comment:
You can nest all the student endpoints inside a parent student url https://codecraft.tv/courses/angular/routing/nested-routes/